### ISSUE TITLE
Could not Visualize during Training

### ISSUE THREAD

**Issue by A7medRamadan2018** [OpenNMT/OpenNMT-py#1613]
I had run training process for data inside data folder and add -tensorboard for visualization but does not work !!

**Comment by francoishernandez** [comment]
Please post your command line as well as the error trace you get.

**Comment by francoishernandez** [comment]
> after training finished, no visualization appeared !!
Do you launch the tensorboard instance properly?

**Comment by A7medRamadan2018** [comment]
I have just add -tensorboard only and wait for visualize

**Comment by francoishernandez** [comment]
You need to execute tensorboard on your logdir.

`tensorboard --logdir <your_log_dir>`

**Comment by A7medRamadan2018** [comment]
I called `onmt_train -h`
usage: onmt_train [-h] [-config CONFIG] [-save_config SAVE_CONFIG]
                  [--src_word_vec_size SRC_WORD_VEC_SIZE]
                  [--tgt_word_vec_size TGT_WORD_VEC_SIZE]
                  [--word_vec_size WORD_VEC_SIZE] [--share_decoder_embeddings]
                  [--share_embeddings] [--position_encoding]
                  [--feat_merge {concat,sum,mlp}]
                  [--feat_vec_size FEAT_VEC_SIZE]
                  [--feat_vec_exponent FEAT_VEC_EXPONENT]
                  [--model_type {text,img,audio,vec}]
                  [--model_dtype {fp32,fp16}]
                  [--encoder_type {rnn,brnn,mean,transformer,cnn}]
                  [--decoder_type {rnn,transformer,cnn}] [--layers LAYERS]
                  [--enc_layers ENC_LAYERS] [--dec_layers DEC_LAYERS]
                  [--rnn_size RNN_SIZE] [--enc_rnn_size ENC_RNN_SIZE]
                  [--dec_rnn_size DEC_RNN_SIZE]
                  [--audio_enc_pooling AUDIO_ENC_POOLING]
                  [--cnn_kernel_width CNN_KERNEL_WIDTH]
                  [--input_feed INPUT_FEED] [--bridge]
                  [--rnn_type {LSTM,GRU,SRU}] [--brnn]
                  [--context_gate {source,target,both}]
                  [--global_attention {dot,general,mlp,none}]
                  [--global_attention_function {softmax,sparsemax}]
                  [--self_attn_type SELF_ATTN_TYPE]
                  [--max_relative_positions MAX_RELATIVE_POSITIONS]
                  [--heads HEADS] [--transformer_ff TRANSFORMER_FF]
                  [--aan_useffn] [--copy_attn]
                  [--copy_attn_type {dot,general,mlp,none}]
                  [--generator_function {softmax,sparsemax}]
                  [--copy_attn_force] [--reuse_copy_attn]
                  [--copy_loss_by_seqlength] [--coverage_attn]
                  [--lambda_coverage LAMBDA_COVERAGE]
                  [--loss_scale LOSS_SCALE] [--apex_opt_level {O0,O1,O2,O3}]
                  --data DATA [--data_ids DATA_IDS [DATA_IDS ...]]
                  [--data_weights DATA_WEIGHTS [DATA_WEIGHTS ...]]
                  [--save_model SAVE_MODEL]
                  [--save_checkpoint_steps SAVE_CHECKPOINT_STEPS]
                  [--keep_checkpoint KEEP_CHECKPOINT]
                  [--gpuid [GPUID [GPUID ...]]]
                  [--gpu_ranks [GPU_RANKS [GPU_RANKS ...]]]
                  [--world_size WORLD_SIZE] [--gpu_backend GPU_BACKEND]
                  [--gpu_verbose_level GPU_VERBOSE_LEVEL]
                  [--master_ip MASTER_IP] [--master_port MASTER_PORT]
                  [--queue_size QUEUE_SIZE] [--seed SEED]
                  [--param_init PARAM_INIT] [--param_init_glorot]
                  [--train_from TRAIN_FROM]
                  [--reset_optim {none,all,states,keep_states}]
                  [--pre_word_vecs_enc PRE_WORD_VECS_ENC]
                  [--pre_word_vecs_dec PRE_WORD_VECS_DEC]
                  [--fix_word_vecs_enc] [--fix_word_vecs_dec]
                  [--batch_size BATCH_SIZE] [--batch_type {sents,tokens}]
                  [--pool_factor POOL_FACTOR] [--normalization {sents,tokens}]
                  [--accum_count ACCUM_COUNT [ACCUM_COUNT ...]]
                  [--accum_steps ACCUM_STEPS [ACCUM_STEPS ...]]
                  [--valid_steps VALID_STEPS]
                  [--valid_batch_size VALID_BATCH_SIZE]
                  [--max_generator_batches MAX_GENERATOR_BATCHES]
                  [--train_steps TRAIN_STEPS] [--single_pass]
                  [--epochs EPOCHS] [--early_stopping EARLY_STOPPING]
                  [--early_stopping_criteria [EARLY_STOPPING_CRITERIA [EARLY_STOPPING_CRITERIA ...]]]
                  [--optim {sgd,adagrad,adadelta,adam,sparseadam,adafactor,fusedadam}]
                  [--adagrad_accumulator_init ADAGRAD_ACCUMULATOR_INIT]
                  [--max_grad_norm MAX_GRAD_NORM]
                  [--dropout DROPOUT [DROPOUT ...]]
                  [--attention_dropout ATTENTION_DROPOUT [ATTENTION_DROPOUT ...]]
                  [--dropout_steps DROPOUT_STEPS [DROPOUT_STEPS ...]]
                  [--truncated_decoder TRUNCATED_DECODER]
                  [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2]
                  [--label_smoothing LABEL_SMOOTHING]
                  [--average_decay AVERAGE_DECAY]
                  [--average_every AVERAGE_EVERY]
                  [--learning_rate LEARNING_RATE]
                  [--learning_rate_decay LEARNING_RATE_DECAY]
                  [--start_decay_steps START_DECAY_STEPS]
                  [--decay_steps DECAY_STEPS]
                  [--decay_method {noam,noamwd,rsqrt,none}]
                  [--warmup_steps WARMUP_STEPS] [--report_every REPORT_EVERY]
                  [--log_file LOG_FILE]
                  [--log_file_level {CRITICAL,WARNING,NOTSET,DEBUG,INFO,ERROR,50,30,0,10,20,40}]
                  [--exp_host EXP_HOST] [--exp EXP] **[--tensorboard]
                  [--tensorboard_log_dir TENSORBOARD_LOG_DIR]**
                  [--sample_rate SAMPLE_RATE] [--window_size WINDOW_SIZE]
                  [--image_channel_size {3,1}]
there are parameters knowed by opennmt 
when I changed **-tensorboard** to  **tensorboard --logdir "my_tensorboard"**
My log is :  **onmt_train: error: argument --gpu_ranks/-gpu_ranks: invalid int value: 'tensorboard'**

**Comment by francoishernandez** [comment]
Please do some research.
Tensorboard is a separate tool. OpenNMT-py only dumps logs that can be read by Tensorboard.

**Comment by A7medRamadan2018** [comment]
yes i understand, many thanks for you
